packages/cozy-intent/src/api/services/NativeService.ts

```diff
@@ -1,45 +1,48 @@
-import { ParentHandshake } from 'post-me'
+import { Connection, ParentHandshake } from 'post-me'
 
 import { MessengerRegister } from '../models/messengers'
 import { NativeEvent } from '../models/events'
 import { NativeMessenger } from '../services/NativeMessenger'
 import { NativeMethodsRegister } from '../models/methods'
+import { StaticService } from './StaticService'
+import { Strings } from '../constants'
 import { WebviewRef } from '../models/environments'
-import { Numbers, Strings } from '../constants'
 
 export class NativeService {
   private messengerService = NativeMessenger
 
+  private localMethods: NativeMethodsRegister
   private messengerRegister = {} as MessengerRegister
 
-  public registerWebview = async (
-    webviewRef: WebviewRef,
-    localMethods: NativeMethodsRegister
-  ): Promise<void> => {
-    const uri = new URL(webviewRef.props.source.uri).hostname
+  constructor(localMethods: NativeMethodsRegister) {
+    this.localMethods = localMethods
+  }
 
-    if (this.messengerRegister[uri]) return
+  public registerWebview = (webviewRef: WebviewRef): void => {
+    const uri = StaticService.getUri(webviewRef)
 
-    const messenger = new this.messengerService(webviewRef)
+    if (this.messengerRegister[uri]) return
 
     this.messengerRegister = {
       ...this.messengerRegister,
-      [uri]: { messenger }
+      [uri]: { messenger: new this.messengerService(webviewRef) }
     }
-
-    await ParentHandshake(
-      messenger,
-      localMethods,
-      Numbers.maxAttempts,
-      Numbers.attemptsInterval
-    )
   }
 
-  public tryEmit = (event: NativeEvent): void => {
-    if (!event.nativeEvent.data.includes(Strings.postMeSignature)) return
+  private initWebview = async (
+    messenger: NativeMessenger
+  ): Promise<Connection> => await ParentHandshake(messenger, this.localMethods)
+
+  public tryEmit = async (event: NativeEvent): Promise<void> => {
+    if (!StaticService.isPostMeMessage(event)) return
+
+    const { message, uri } = StaticService.parseNativeEvent(event)
 
-    this.messengerRegister[
-      new URL(event.nativeEvent.url).hostname
-    ].messenger.onMessage(event)
+    if (message === Strings.webviewIsRendered)
```

Comment:
I don't understand the PR without the modification of the caller... 

It seems that the caller will send an event "webviewIsRendered" and then we start the handshake process. Right ? 

When are you going to send this message from the caller?

Reply:
it is sent here https://github.com/cozy/cozy-libs/pull/1417/files#diff-416061f3de64eb10ebdfdbe98f8e86fcfbcb2f177e58e82e2462f009d64c4f6eR31

unless I misunderstood your question

Reply:
Nice, thank you! This is exactly what I was looking for. 

So we launch the "childHandshake" at the same time than the "ParentHandshake". Is it problematic? Or is it want we exactly want?

Reply:
Well, childhandshake promise should be launched first, and Parent some milliseconds after, though I'm not sure if there are some cases where post-me would fail the handshake with this configuration. Post-me documentation isn't very clear on that... 

childhandshake has no timeout, that means for me it should be launched first, because it's waiting for the parent to ping it

Reply:
@doubleface can you share you expertise with us on this subject? :)

Reply:
https://github.com/alesgenova/post-me/blob/main/packages/core/src/handshake.ts

for me, childhandshake is waiting forever for the parent ping. So it should be ok if it's always init before parent (which is the case here, since we send a message in a sync way). Which means the parent should find the children at the first ping